dashboard/test/controllers/api/v1/pd/workshop_survey_foorm_report_controller_test.rb

```diff
@@ -147,17 +155,78 @@ class WorkshopSurveyFoormReportControllerTest < ::ActionController::TestCase
       assert_equal 5, overall_facilitator_rollup_facilitator[csf_workshop_1_facilitator][:response_count]
     end
 
-    def create_surveys_for_csf_workshop(csf_workshop, facilitator_id, high_count, low_count)
-      create_list :csf_intro_post_facilitator_workshop_submission,
-        low_count,
-        :answers_low,
-        pd_workshop_id: csf_workshop.id,
-        facilitator_id: facilitator_id
-      create_list :csf_intro_post_facilitator_workshop_submission,
-        high_count,
-        :answers_high,
-        pd_workshop_id: csf_workshop.id,
-        facilitator_id: facilitator_id
+    test 'return unauthorized for unauthorized users' do
+      generic_teacher = create :teacher
+      other_facilitator = create :facilitator
+      program_manager = create :program_manager
+      workshop_organizer = create :workshop_organizer
+      csf_workshop = create :csf_workshop,
+        started_at:  Time.now.utc - 1.day,
+        ended_at: Time.now.utc - 1.hour
+
+      this_facilitator = csf_workshop.facilitators[0]
+
+      expected_authorization = [
+        {user: this_facilitator, expected_response: :success, type: "facilitator for this workshop"},
+        {user: program_manager, expected_response: :success, type: "program manager"},
+        {user: workshop_organizer, expected_response: :success, type: "workshop organizer"},
+        {user: @workshop_admin, expected_response: :success, type: "workshop admin"},
+        {user: generic_teacher, expected_response: :unauthorized, type: "teacher"},
+        {user: other_facilitator, expected_response: :unauthorized, type: "other facilitator"}
+      ]
+
+      expected_authorization.each do |data|
+        sign_in data[:user]
+        get :generic_survey_report, params: {workshop_id: csf_workshop.id}
+        assert_response data[:expected_response], "#{data[:type]} had an unexpected result"
+        sign_out data[:user]
+      end
+    end
+
+    test 'filters facilitator data if facilitator is signed in' do
+      csf_workshop = create :csf_workshop,
+        started_at:  Time.now.utc - 1.day,
+        ended_at: Time.now.utc - 1.hour,
+        num_facilitators: 2
+      facilitator_1 = csf_workshop.facilitators[0]
+      facilitator_2 = csf_workshop.facilitators[1]
+      create_surveys_for_csf_workshop(csf_workshop, csf_workshop.facilitators.pluck(:id), 5, 2)
```

Comment:
It would be great if the test somewhere explained what these parameters (especially `high_count` and `low_count`) are.

Reply:
added comment on method